ams/ams-server/src/main/java/com/netease/arctic/ams/server/controller/CatalogController.java

```diff
@@ -0,0 +1,427 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.ams.server.controller;
+
+import com.netease.arctic.ams.api.CatalogMeta;
+import com.netease.arctic.ams.api.properties.CatalogMetaProperties;
+import com.netease.arctic.ams.api.properties.TableFormat;
+import com.netease.arctic.ams.server.config.ConfigFileProperties;
+import com.netease.arctic.ams.server.controller.response.ErrorResponse;
+import com.netease.arctic.ams.server.controller.response.OkResponse;
+import com.netease.arctic.ams.server.model.CatalogRegisterInfo;
+import com.netease.arctic.ams.server.model.CatalogSettingInfo;
+import com.netease.arctic.ams.server.model.CatalogSettingInfo.ConfigFileItem;
+import com.netease.arctic.ams.server.service.IMetaService;
+import com.netease.arctic.ams.server.service.ServiceContainer;
+import com.netease.arctic.ams.server.service.impl.CatalogMetadataService;
+import com.netease.arctic.ams.server.service.impl.PlatformFileInfoService;
+import io.javalin.http.Context;
+import io.javalin.http.HttpCode;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.netease.arctic.ams.api.properties.CatalogMetaProperties.CATALOG_TYPE_CUSTOM;
+
+public class CatalogController extends RestBaseController {
+  private static final Logger LOG = LoggerFactory.getLogger(CatalogController.class);
+  private static final IMetaService iMetaService = ServiceContainer.getMetaService();
+  private static CatalogMetadataService catalogMetadataService = ServiceContainer.getCatalogMetadataService();
+  private static PlatformFileInfoService platformFileInfoService = ServiceContainer.getPlatformFileInfoService();
+
+  private static final String AUTH_CONFIG_KEY_TYPE = "auth_config.type";
```

Comment:
Could we reuse the same property keys in `CatalogMetaProperties` ?

Reply:
> Could we reuse the same property keys in `CatalogMetaProperties` ?

done